File: apps/mybookkeeper/backend/app/schemas/properties/utility_offer_search_response.py
"""Response for the better-plan search across every property."""
from __future__ import annotations

from pydantic import BaseModel

from app.schemas.properties.utility_offer_group import UtilityOfferGroup


class UtilityOfferSearchResponse(BaseModel):
    groups: list[UtilityOfferGroup] = []
    # Reference consumption the savings figures were computed at, so the client
    # can label them honestly rather than implying a bill-specific promise.
    reference_annual_kwh: int
    # True when at least one group came back with offers.
    has_any_offers: bool = False

File: apps/mybookkeeper/frontend/src/shared/types/utility/utility-offer-search-response.ts
import type { UtilityOfferGroup } from "@/shared/types/utility/utility-offer-group";

/**
 * Mirrors `backend/app/schemas/properties/utility_offer_search_response.py`.
 */
export interface UtilityOfferSearchResponse {
  groups: UtilityOfferGroup[];
  /**
   * Usage the savings figures were computed at. Shown to the operator so a
   * saving reads as "at this reference usage", not as a promise about a bill.
   */
  reference_annual_kwh: number;
  has_any_offers: boolean;
}
